spec/models/time_range_spec.rb

```diff
@@ -38,6 +38,45 @@
     end
   end
 
+  describe "duration_days" do
+    subject(:duration_days) { range.duration_days }
+
+    describe "when start_at and end_at are nil" do
+      let(:start_at) { nil }
+      let(:end_at) { nil }
+
+      it { is_expected.to be_nil }
+    end
+
+    describe "when start_at is nil and end_at is not" do
+      let(:start_at) { nil }
+      let(:end_at) { Time.zone.now }
+
+      it { is_expected.to be_nil }
+    end
+
+    describe "when end_at is nil and start_at is not" do
+      let(:start_at) { Time.zone.now }
+      let(:end_at) { nil }
+
+      it { is_expected.to be_nil }
+    end
+
+    describe "when it starts and end the following day less than 24 hours appart" do
+      let(:start_at) { Time.zone.local(2024, 10, 24, 16, 0, 0) }
+      let(:end_at) { Time.zone.local(2024, 10, 25, 11, 0, 0) }
+
+      it { is_expected.to eq(1) }
+    end
+
+    describe "when it starts and end the following day more than 24 hours appart" do
+      let(:start_at) { Time.zone.local(2024, 10, 24, 10, 0, 0) }
+      let(:end_at) { Time.zone.local(2024, 10, 25, 19, 0, 0) }
+
+      it { is_expected.to eq(2) }
+    end
```

Comment:
I'm not sure what's best here. We're not supposed to have these cases. For us, a daily reservation can only use 24-hour blocks. I'm not even sure what the expected math is here.

Reply:
This is just testing the `TimeRange` which is fairly it's own abstraction I think it's better if we have a defined behavior for other than expected business logic data.

Reply:
I see your point. I agree that the current implementation works this way and it's usually good to have specs consider different (or all) cases.

For me, the problem is that this case is invalid for us*, so covering it creates an idea of "expected behavior" that isn't actually expected. If somehow we tried to calculate how many days the reservation lasted and it wasn't a natural number, I'd much rather get an error immediately than have clients ask why billing is working in an unexpected way.

*Universities can't charge users for longer than what they had the instrument available for them, so 36 hours can't be 2 days. This is the reason behind having strict 24-hour blocks.

Reply:
I'll just move the ceil call over to the reservation class, where duration_days is used. This value is computed out of validated data so if it returns an incorrect value it should have been captured elsewhere